search-endpoint.md

```diff
@@ -0,0 +1,397 @@
+# Discovery Search Endpoints
+[Back to the list of all defined endpoints](endpoints.md)
+
+## Search Endpoint
+**/api/discover/search**   
+
+This endpoint provides the functionality of the Discovery search screen: http://demo.dspace.org/xmlui/discover.
+
+Provide access to the Discovery search system (SOLR based). This endpoint returns the list of available DSpace object (DSO) types that can be searched:
+* all: Execute a query over all available DSO types.
+* items: Only search the DSpace items.
+* communities: Search within the DSpace community records.
+* collections: Limit the search to DSpace collection records.
+
+Example: TODO
+
+### Search with a given DSO type
+**/api/discover/search/<:dso-type>**
```

Comment:
as this is not really an endpoint that expose resources I think it will be easier to write your own controller instead to stress yourself in the RestResourceController contract. You should be able to do that overriding the mapping using the Order annotation or a more precise match than the one used by the RestResourceController, see https://github.com/DSpace/DSpace/blob/master/dspace-spring-rest/src/main/java/org/dspace/app/rest/BitstreamContentRestController.java#L35 for an example